superdesk/core/resources/fields.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2024 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from typing_extensions import TypeVar, Generic, ClassVar, Dict, Any, cast, Type
+
+from datetime import datetime
+from pydantic_core import core_schema
+from pydantic import (
+    GetJsonSchemaHandler,
+    GetCoreSchemaHandler,
+    WithJsonSchema,
+    Field,
+    ConfigDict,
+)
+from pydantic.json_schema import JsonSchemaValue
+from pydantic.dataclasses import dataclass
+from bson import ObjectId as BsonObjectId
+
+DefaultModelConfig = ConfigDict(
+    arbitrary_types_allowed=True,
+    validate_assignment=True,  # Revalidate on field assignment
+    populate_by_name=True,
+)
+
+
+def get_core_schema_from_type(class_type: Type) -> core_schema.CoreSchema:
+    if class_type == str:
+        return core_schema.str_schema()
+    elif class_type == int:
+        return core_schema.int_schema()
+    elif class_type == float:
+        return core_schema.float_schema()
+    elif class_type == bool:
+        return core_schema.bool_schema()
+    elif class_type == bytes:
+        return core_schema.bytes_schema()
+    elif class_type == datetime:
+        return core_schema.datetime_schema()
+    raise RuntimeError(f"Unsupported base class type: {class_type}")
```

Comment:
May I suggest an alternative version to improve readability. What do you think?

```suggestion
def get_core_schema_from_type(class_type: Type) -> core_schema.CoreSchema:
    schema_mapping = {
        str: core_schema.str_schema,
        int: core_schema.int_schema,
        float: core_schema.float_schema,
        bool: core_schema.bool_schema,
        bytes: core_schema.bytes_schema,
        datetime: core_schema.datetime_schema,
    }

    schema_function = schema_mapping.get(class_type)
    if schema_function:
        return schema_function()
    
    raise RuntimeError(f"Unsupported base class type: {class_type}")
```